src/Tasks/ResolveSDKReference.cs

```diff
@@ -1251,7 +1259,7 @@ private void CreateResolvedReferenceItem(string targetConfiguration, string targ
                     AddResolutionWarning("ResolveSDKReference.MaxPlatformVersionNotSpecified", projectName, DisplayName, Version, targetPlatformIdentifier, targetPlatformVersionFromItem.ToString(), targetPlatformIdentifier, targetPlatformVersion.ToString());
                 }
 
-                if (!String.IsNullOrEmpty(TargetPlatform) && !String.Equals(targetPlatformIdentifier, TargetPlatform))
+                if (!String.IsNullOrEmpty(TargetPlatform) && !String.Equals(targetPlatformIdentifier, TargetPlatform) && (!platformAliases.TryGetValue(TargetPlatform, out string platform) || !String.Equals(targetPlatformIdentifier, platform, StringComparison.OrdinalIgnoreCase)))
```

Comment:
```suggestion
                if (!String.IsNullOrEmpty(TargetPlatform) && (!String.Equals(targetPlatformIdentifier, TargetPlatform) || (platformAliases.TryGetValue(TargetPlatform, out string platform) && !String.Equals(targetPlatformIdentifier, platform, StringComparison.OrdinalIgnoreCase))))
```

This reads easier to me. Even if this isn't preferred: If trygetvalue returned false, why bother checking if platform == TPI?

Reply:
If TargetPlatform is not null or empty, and it isn't the same as targetPlatformIdentifier, and it isn't the same as the aliased version, we should fail.

Your version says that if TargetPlatform is not null or empty, and it _is_ the same as targetPlatformIdentifier, but it's also in the dictionary and the aliased version is not the same, then we should fail, and that isn't true. Similarly, with your version, if TargetPlatform is not null or empty, and it's different from targetPlatformIdentifier, then we don't bother checking the dictionary, and we fail, which again is not what we want.

Reply:
AH my bad, this is the **failing case**, not the continuing case.

Reply:
Added a unit test 🙂